### PR TITLE
chore: fix the method signature for pruneAllImages of Podman

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?

the method accepts a boolean for dangling

https://github.com/podman-desktop/podman-desktop/blob/7da6e4a48972e27a0eab4ae06580a7a57f492cb7/packages/main/src/plugin/dockerode/libpod-dockerode.ts#L379

 but the implementation is ignoring the parameter value
make it handling the value

### What issues does this PR fix or reference?

related to https://github.com/podman-desktop/podman-desktop/issues/7043

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/dockerode/libpod-dockerode.ts [hunk 1]
```diff
@@ -530,18 +530,22 @@ export class LibpodDockerode {
     };
 
     // add pruneAllImages
-    prototypeOfDockerode.pruneAllImages = function (): Promise<unknown> {
+    prototypeOfDockerode.pruneAllImages = function (all: boolean): Promise<unknown> {
       const optsf = {
-        path: '/v4.2.0/libpod/images/prune?all=true&', // this works
-        // For some reason the below doesn't work
-        // options: {all: 'true'}, // this doesn't work
+        path: '/v4.2.0/libpod/images/prune?',
```
Comment (jeffmaury) [inline packages/main/src/plugin/dockerode/libpod-dockerode.ts]:
```suggestion
        path: '/v4.2.0/libpod/images/prune',
```

File: packages/main/src/plugin/dockerode/libpod-dockerode.ts [hunk 2]
```diff
@@ -530,18 +530,22 @@ export class LibpodDockerode {
...
         method: 'POST',
         statusCodes: {
           200: true,
           400: 'bad parameter',
           500: 'server error',
         },
       };
+      if (all) {
+        optsf.path += 'all=true&';
```
Comment (jeffmaury) [inline packages/main/src/plugin/dockerode/libpod-dockerode.ts]:
```suggestion
        optsf.path += '?all=true&';
```

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]

**Review by SoniaSandler — APPROVED** [review]
LGTM codewise

**Pull request merged**